File: src/scripts/remote_system_info.py
#/usr/bin/python3
import os, platform
import psutil
import datetime
import json

boot_time_timestamp = psutil.boot_time()
bt = datetime.datetime.fromtimestamp(boot_time_timestamp)
uptime = datetime.datetime.now() - bt
uptime_seconds = uptime.total_seconds()
data = {
  "homeFolder": os.path.expanduser("~"),
  "type": os.name,
  "release": platform.release(),
  "platform": platform.system(),
  "pythonVersion": platform.python_version(),
  "cpuArchitecture": platform.machine(),
  "totalMemory": os.sysconf("SC_PAGE_SIZE") * os.sysconf("SC_PHYS_PAGES"),
  "freeMemory": os.sysconf("SC_PAGE_SIZE") * os.sysconf("SC_AVPHYS_PAGES"),
  "uptime": uptime_seconds,
  "currentFolder": os.getcwd()
}
json_response = json.dumps(data)
print(json_response)

File: src/scripts/remote_system_info.js
const os = require('os');
const fs = require('fs');
const path = require('path');

// Function to safely get system information
function getInfo() {
  try {
    const uptime = os.uptime();
    const totalMemory = os.totalmem();
    const freeMemory = os.freemem();
    
    const systemInfo = {
      homeFolder: os.homedir(),
      type: os.type(),
      release: os.release(),
      platform: os.platform(),
      architecture: os.arch(),
      totalMemory: totalMemory,
      freeMemory: freeMemory,
      uptime: uptime,
      currentFolder: process.cwd()
    };

    return JSON.stringify(systemInfo, null, 2);
  } catch (error) {
    return JSON.stringify({ error: error.message });
  }
}

// Write the information to stdout
console.log(getInfo());

